src/DonorProfiles/Tabs/DonationHistoryTab/DonationsRoute.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Give\DonorProfiles\Tabs\DonationHistoryTab;
+
+use WP_REST_Request;
+use Give\DonorProfiles\Tabs\Contracts\Route as RouteAbstract;
+use Give\DonorProfiles\Repositories\Donations as DonationsRepository;
+
+/**
+ * @since 2.10.0
+ */
+class DonationsRoute extends RouteAbstract {
+
+	/** @var string */
+	public function endpoint() {
+		return 'donations';
+	}
+
+	public function args() {
+		return [];
+	}
+
+	/**
+	 * @param WP_REST_Request $request
+	 *
+	 * @return array
+	 *
+	 * @since 2.10.0
+	 */
+	public function handleRequest( WP_REST_Request $request ) {
+		return $this->getData();
+	}
+
+	/**
+	 * @return array
+	 *
+	 * @since 2.10.0
+	 */
+	protected function getData() {
+
+		$repository = new DonationsRepository();
```

Comment:
This could be injected as a dependency and resolved from the container. One benefit would be to mock the repository for unit testing the return format of the `getData()` method.

Reply:
I've gone ahead and changed this so that the DonationsRepository is injected as a dependency. I'm holding off on registering the repository as a global, since for the time being this is the only place it is referenced. Done, ref: https://github.com/impress-org/givewp/pull/5577/commits/4677b3f9d697f820499e91800c719f076ebac52e

Reply:
To clarify, I was suggesting that `DonorRepository` be resolved from the container via the class constructor, which eliminates the need for manually instantiating a class.